### PR TITLE
Specific Subnet and VPC on initial deployment

### PR DESCRIPTION
**Pull request by root-beer**
Ref: https://skyarch.backlog.jp/view/NC-672

## TODO

- [x] Modify CFTemplates
- [x] Modify input forms

## Update Process
```
bundle exec rake i18n:js:export
```

### CODE REVIEW

File: app/assets/javascripts/app_settings.js [hunk 1]
```diff
@@ -59,10 +60,10 @@
   };
 
 
-  var is_fill_input = function() {
-    var set = get_settings();
-    for (var i in set) {
-      if (set[i] === '') {
+  var is_fill_required_input = function () {
+    var elements = document.querySelectorAll(required_inputs);
+    for (var el of elements.values()) {
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
* 'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: app/controllers/app_settings_controller.rb [hunk 2]
```diff
@@ -28,6 +30,15 @@ def create
     keypair_name      = settings.delete(:keypair_name)
     keypair_value     = settings.delete(:keypair_value)
 
+    vpc_id    = settings.delete(:vpc_id)    || ''
+    subnet_id = settings.delete(:subnet_id) || ''
+
+    cf_params = {
+      VpcId:    vpc_id,
+      SubnetId: subnet_id
```
Comment (skyarch-networks) [inline app/controllers/app_settings_controller.rb]: __[rubocop]__
* Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline hash. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request
[Auto Correct by SideCI](https://sideci.com/gh/31754199/issues/pull_request/372/diff/331bbfc88c04a9aee36e33b55b6f1068e8afbe19?open_modal=1#rubocop)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by root-beer** [comment]
Please review and merge PR. @Joeper214 :bow:

**Comment by Joeper214** [comment]
I'll review now.

**Comment by Joeper214** [comment]
There's a problem I encountered while testing the feature.

The log shows the error message.
screen shot 2017-03-23 at 16 12 39

But it doesn't reflect in the application.
screen shot 2017-03-23 at 16 12 45


Is it possible if we can add verification for the `VPC ID` and `Subnet ID` Verification during submission? 

I guess I already did  some verification features like this when I developed the marketplace branch.

**Comment by Joeper214** [comment]
Like this.

https://github.com/skyarch-networks/skyhopper/blob/aws-marketplace/app/controllers/app_settings_controller.rb#L72-L84

**Comment by Joeper214** [comment]
And Maybe we should only verify these fields if the input is not `nil`

**Comment by root-beer** [comment]
Ready for review. :bow:

**Comment by Joeper214** [comment]
かしこまりました。I'll review now.

**Comment by Joeper214** [comment]
I tried to enter these inputs in the optional parameters(For testing purposes).

VPC ID: `hogehoge`
Subnet ID: `fugafuga`

screen shot 2017-04-27 at 16 10 01


But it failed to check whether these ID's do exists.
screen shot 2017-03-23 at 16 12 39


And no error message was returned.
screen shot 2017-03-23 at 16 12 45


What do you think?

**Comment by Joeper214** [comment]
Don't we need to verify if these ID's exists?

**Comment by root-beer** [comment]
I got it. I'll add verification.

**Comment by Joeper214** [comment]
ようろしくおねがいします。:bow:

File: app/controllers/app_settings_controller.rb [hunk 3]
```diff
@@ -152,22 +171,33 @@ def build_ws_message(status, msg = nil)
...
   class EIPLimitError < StandardError; end
 
   # @param [String] region
   # @param [String] access_key_id
   # @param [String] secret_access_key
   # @raise [EIPLimitError] raise error when cann't allocate EIP.
-  def check_eip_limit!(region, access_key_id, secret_access_key)
-    e = Aws::EC2::Client.new(region: region, access_key_id: access_key_id, secret_access_key: secret_access_key)
-    a = e.describe_account_attributes
+  def check_eip_limit!
```
Comment (skyarch-networks) [inline app/controllers/app_settings_controller.rb]: __[method_definition_validator]__
* Lint/DefinitionValidator: Too many arguments at [spec/controllers/app_settings_controller_spec.rb#L131](/skyarch-networks/skyhopper/blob/c32ec9b5a86d7f3fabd008fd41819dd841f60292/spec/controllers/app_settings_controller_spec.rb#L131)

**Comment by root-beer** [comment]
Ready for review. 🙇

**Comment by Joeper214** [comment]
I'll Review now. 👍

**Comment by Joeper214** [comment]
Can we use native javascript [here](https://github.com/skyarch-networks/skyhopper/blob/0334876a734cf6ded8ee1020d3b1de02ef0c1fab/app/assets/javascripts/app_settings.js#L37-L39
)
 instead of lodash?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape

**Comment by root-beer** [comment]
It does not become the expected display because the logic is different.

**Comment by Joeper214** [comment]
かしこまりました。

**Comment by Joeper214** [comment]
LGTM Thanks! 🙇

**Comment by root-beer** [comment]
ありがとう！:bow:

**Pull request merged**